TAS: update doc for GPUs (#6912)
kubernetes-sigs/kueue: site/content/en/docs/concepts/topology_aware_scheduling.md

{{< include "examples/tas/sample-queues.yaml" "yaml" >}}

> An example for managing GPUS:

Reviewer: ```suggestion
An example for managing GPUs:
```
nit

	indicated by the annotation value (e.g. within a rack or within a block).
- `kueue.x-k8s.io/podset-unconstrained-topology` - indicates that a PodSet requires
    Topology Aware Scheduling, and requires scheduling all pods on any nodes without

>     topology considerations. In other words, this considers if all pods could be accommodated 

Reviewer: Please revert the other changes in the file in the PR. They look sensible, but submit separetely. I think they could conflict otherwise.

### Hot swap support
{{% alert title="Note" color="primary" %}}
To enable the feature, you have to set the [feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
`TASFailedNodeReplacement` to `true` and the last element of the nodeLabels in

> Topology has to be `kubernetes.io/hostname`. This feature was introduced to

Author: ```suggestion
`kubernetes.io/hostname`. This feature was introduced to Kueue in version 0.12.
```

> Kueue in version 0.12.

Author: ```suggestion
```

Currently, there are limitations for the compatibility of TAS with other
features, including:
- Some scheduling directives (e.g. pod affinities and anti-affinities) are ignored.
- The "podset-required-topology" annotation may fail if the underlying
  ClusterAutoscaler cannot provision nodes that satisfy the domain constraint.

> - A ClusterQueue for [MultiKueue](multikueue.md) referencing a ResourceFlavor

Author: ```suggestion
- a ClusterQueue for [MultiKueue](multikueue.md) referencing a ResourceFlavor
```
